libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -179,6 +180,57 @@ bool IRGeneratorForStatements::visit(FunctionCall const& _functionCall)
 	return false;
 }
 
+bool IRGeneratorForStatements::visit(InlineAssembly const& _inlineAsm)
+{
+	auto const& loc = _inlineAsm.location();
+
+	solAssert(loc.start != 1 && loc.end != -1, "Expected source location");
+
+	vector<string> params, retParams, args;
+	vector<map<string, string>> assignVars;
+	// Prepare parameters and arguments
+	for (auto const& pair: _inlineAsm.annotation().externalReferences)
+	{
+		auto varDecl = dynamic_cast<VariableDeclaration const*>(pair.second.declaration);
+		solUnimplementedAssert(!!varDecl, "");
+
+		args.emplace_back(m_context.variableName(*varDecl));
+		params.emplace_back(varDecl->name());
+		retParams.emplace_back("ret_" + varDecl->name());
+		assignVars.emplace_back(
+			map<string, string>{{"retParam", retParams.back()},{"param", params.back()}}
+		);
+	}
+
+	// Skip the "assembly" literal
+	int blockStart = loc.start + string("assembly").size();
+
+	stringstream funcName;
+
+	funcName <<	"inline_assembly_" << loc.start << "_" << loc.end;
+
+	Whiskers inlineAsm(R"(
+		function <funcName> (<params>) -> <retParams> {
+			<asmBlock>
+<#assignVars>
```

Comment:
Indentation ;-).

Reply:
this was actually deliberate so that it would have nice indentation in the generated code

Reply:
Hm... haven't thought about that... not sure which has precedence in that case - but I think it's rather the generating and not the generated code ;-).

Reply:
I agree with @ekpyron here :p

Reply:
fiiiineeee